content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
+For example, while you’re typing on your computer. You’re only interested in punching the keys believing that it produces results on your screen but you know nothing about the mechanism that produces the result and I’m sure you haven’t even thought about it yet. Have you? 
+Another example to make things clearer,  imagine a car driver, he is only interested in starting, moving, and stopping the car. Accelerate or change gear and brake but he is not interested in the mechanism that works internally to perform all the above functions. Simply, in abstraction, the implementation details are hidden from the user while the functionality is exposed to the user. 
+The while the user will only know what a function(method in java) does but not how it does it.
 
 How to implement Abstraction in Java
+
 In Java, abstraction is implemented in either of these two ways
--[1]: by using the non-access modifier ‘abstract’ keyword. 
--[2]: Interface
 
+- By using the non-access modifier ‘abstract’ keyword. 
+
+- Interface
+
+- Non-Access Modifier keyword “Abstract” Implementation
 
-- 1: Non-Access Modifier keyword “Abstract” Implementation
 The abstract access modifier keyword can only be used with classes, methods but not with variables. Methods with the keyword abstract are methods without implementation.
 
-To implement an abstract class in Java, use the abstract keyword immediately before the class keyword preceding the class name e.g public abstract class Circle, where Circle is the class name. Use this pattern for abstract method implementation, public abstract returnType methodName() i.e public abstract double getArea().
+To implement an abstract class in Java, use the abstract keyword e.g public abstract class Circle, where Circle is the class name. 
+Use this pattern for abstract method implementation, public abstract returnType methodName() i.e public abstract double getArea().
 
 Code snippet
+
 ```java
+
 public abstract class Dog {
-	
-	public abstract void run();
 
+   public abstract void run();
 
 }
+
 ```
 
-With the above code snippet, we've created an abstract class and an abstract method but one pertinent feature of abstract classes is that objects cannot be instantiated or created from it.But don’t forget, classes only exist for the purpose of creating an object. To be able to create an object from an abstract class we have to create a new class that inherit from the superclass(base, parent) class by using the extends keyword.
+With the above code snippet, we've created an abstract class and an abstract method but one pertinent feature of abstract class is that objects cannot be instantiated or created from it. But don’t forget, classes only exist to create an object. To be able to create an object from an abstract class. We have to create a new class that inherits from the superclass(base, parent) class by using the extends keyword.
```

Comment:
>With the above code snippet, we've created an abstract class and an abstract method but one pertinent feature of abstract class is that objects cannot be instantiated or created from it.

There is a minor grammatical error, and I think it would be best to break this down into two sentences. These would be fixed like so:

> With the above code snippet, we've created an abstract class and an abstract method. One pertinent feature of an abstract class is that objects cannot be instantiated or created from it.

Reply:
> To be able to create an object from an abstract class. We have to create a new class that inherits from the superclass(base, parent) class by using the extends keyword.

The way you broke down the sentence doesn't work grammatically. You cut the flow of the sentence/idea into two sentences in a way that's confusing and changes the meaning. I.e. the sentences: **To be able to create an object from an abstract class.** and **We have to create a new class that inherits from.** should be together. Adding the periods between makes it look like two separate ideas when they are a part of the same phrase/idea. I would fix this part like so:

> To create an object from an abstract class, we need to create a new class that inherits from it. We can do this using the `extends` keyword.